inference-engine/src/transformations/include/transformations/snippets/generator.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <transformations_visibility.hpp>
+#include "snippets_isa.hpp"
+
+namespace ngraph {
+namespace snippets {
+
+using code = const uint8_t *;
+using RegInfo = std::pair<std::vector<size_t>, std::vector<size_t>>;
+
+TRANSFORMATIONS_API auto getRegisters(std::shared_ptr<ngraph::Node>& n) -> ngraph::snippets::RegInfo;
+
+class Emitter {
+public:
+    Emitter(const std::shared_ptr<ngraph::Node>& n) {
+    }
+
+    virtual void emit_code(const std::vector<size_t>& in,
+                           const std::vector<size_t>& out,
+                           const std::vector<size_t>& pool = {},
+                           const std::vector<size_t>& gpr  = {}) const = 0;
+
+    virtual void emit_data() const {
+    }
+};
+
+class TargetMachine {
+public:
+    virtual auto getJitters() -> std::map<const ngraph::DiscreteTypeInfo, std::function<std::shared_ptr<Emitter>(std::shared_ptr<ngraph::Node>)>>{
+        return {};
+    }
+};
+
+class Schedule {
+public:
+    Schedule() : work_size({}), is_flat(false), ptr(nullptr) {}
+    Schedule(const Shape& ws, bool f, code p) : work_size(ws), is_flat(f), ptr(p) {}
+
+    Shape work_size {};
+    bool is_flat {false};
+    code ptr {nullptr};
+};
+
+class TRANSFORMATIONS_API Generator {
+public:
+    Generator() = default;
+    virtual ~Generator() = default;
+
+    // FIXME: generates code for a specific tile decided by whom?
```

Comment:
I think we have a lot of fixme or todo comments. Do you have a plan to fix it before the merge?

Reply:
Removed, most of them are just POC legacy